packages/core/solidity/src/utils/transform-tron.test.ts

```
@@ -0,0 +1,123 @@
+test('preserves the entire program when no TRON-relevant tokens present', t => {
+  // 0.8.20 is below the tron-solc cap, so it stays as-is.
+  const source = 'pragma solidity ^0.8.20;\ncontract Plain { uint256 x; }\n';
+  t.is(rewriteForTron(source), source);
+});
+
+test('caps pragma at 0.8.26 (the current tron-solc maximum)', t => {
+  // Above the cap — downgraded.
+  t.is(
+    rewriteForTron('pragma solidity ^0.8.27;\ncontract Foo {}\n'),
+    'pragma solidity ^0.8.26;\ncontract Foo {}\n',
+  );
```

ESLint

[error] Replace `⏎····rewriteForTron('pragma·solidity·^0.8.27;\ncontract·Foo·{}\n'),⏎····'pragma·solidity·^0.8.26;\ncontract·Foo·{}\n',⏎··` with `rewriteForTron('pragma·solidity·^0.8.27;\ncontract·Foo·{}\n'),·'pragma·solidity·^0.8.26;\ncontract·Foo·{}\n'`

(prettier/prettier)